### PR TITLE
Fixed issue [security] #19146: Store XSS when editing label set

### PR DESCRIPTION
**Pull request by Shnoulle**
Dev: encode in dropdown
Dev: no need HTML in label_name : filtering

### CODE REVIEW

File: application/models/LabelSet.php [hunk 1]
```diff
@@ -286,4 +287,14 @@ public function hasPermission($sPermission, $sCRUD = 'read', $iUserID = null)
...
     }
+
+    /**
+     * Sanitize string for any attribute
+     * @param string $attribute to sanitize
+     * @return string sanitized attribute
+     */
+    public static function sanitizeAttribute($attribute)
+    {
+        return filter_var($attribute, @FILTER_SANITIZE_STRING);
```
Comment (gabrieljenik) [inline application/models/LabelSet.php:298]: What's the `@` for in front of the constant? Never used it like that... 

Also, do we have any other way of sanitizing on ther models? I am realziing, `FILTER_SANITIZE_STRING` is deprectaed. So if using something else somwehere, better use that. right?

Comment (Shnoulle) [reply application/models/LabelSet.php:298]: > Also, do we have any other way of sanitizing on ther models?

No, else i use it. Near copy/paste from Token here.

Comment (Shnoulle) [reply application/models/LabelSet.php:298]: > FILTER_SANITIZE_STRING` is deprectaed.

If i understand right : it's FILTER_FLAG_NO_ENCODE_QUOTES deprecated.

> Encoding quotes can be disabled by setting FILTER_FLAG_NO_ENCODE_QUOTES. (Deprecated as of PHP 8.1.0, use [htmlspecialchars()](https://www.php.net/manual/en/function.htmlspecialchars.php) instead.)

**Review by gabrieljenik — APPROVED** [review]
Code looks OK

File: application/models/LabelSet.php (hunk 1 above)
Comment (gabrieljenik) [reply application/models/LabelSet.php:298]: Look at the changelog section here.
https://www.php.net/manual/en/filter.filters.sanitize.php
image

> If i understand right : it's FILTER_FLAG_NO_ENCODE_QUOTES deprecated.

>> Encoding quotes can be disabled by setting FILTER_FLAG_NO_ENCODE_QUOTES. (Deprecated as of PHP 8.1.0, use [htmlspecialchars()](https://www.php.net/manual/en/function.htmlspecialchars.php) instead.)

It is a bit confusing, but in changelog is clear.

**Review by gabrieljenik — CHANGES_REQUESTED** [review]
Let's keep on reviewing about the sanitize... I am not too sure about inclding something which we know is already deprecated.

**Comment by gabrieljenik** [comment]
> Code looks OK

Sorry, reviewed again.

File: application/models/LabelSet.php (hunk 1 above)
Comment (Shnoulle) [reply application/models/LabelSet.php:298]: A good old preg_replace ?

Comment (Shnoulle) [reply application/models/LabelSet.php:298]: Or a `FILTER_SANITIZE_SPECIAL_CHARS` ?

Comment (Shnoulle) [reply application/models/LabelSet.php:298]: strip_tags ?

Comment (gabrieljenik) [reply application/models/LabelSet.php:298]: hmm whatever is already used in other models for similar situations... :) sanitize _helper?

**Review by gabrieljenik — APPROVED** [review]
Code looks OK

File: application/models/LabelSet.php (hunk 1 above)
Comment (Shnoulle) [reply application/models/LabelSet.php]: > hmm whatever is already used in other models for similar situations... :) sanitize _helper?

There no global santizer validator, only xss, but here : i jyust don't want (care) of HTML, and needed for all user (admin included)

Found only 2 place

- https://github.com/LimeSurvey/LimeSurvey/blob/d01845f9895f4ada9abe31e0b2a5b772468babc5/application/models/Token.php#L322
- https://github.com/LimeSurvey/LimeSurvey/blob/d01845f9895f4ada9abe31e0b2a5b772468babc5/application/models/Token.php#L311

**Comment by Shnoulle** [comment]
All green

**Pull request merged**